go/libraries/doltcore/sqle/enginetest/dolt_queries_merge.go

```diff
@@ -4150,6 +4150,28 @@ var ThreeWayMergeWithSchemaChangeTestScripts = []MergeScriptTest{
 	},
 
 	// Schema conflict test cases
+	{
+		Name: "index conflicts: both sides add an index with the same name, same columns, but different type",
+		AncSetUpScript: []string{
+			"CREATE table t (pk int primary key, col1 int, col2 varchar(100));",
+		},
+		RightSetUpScript: []string{
+			"alter table t add index idx1 (col2(2));",
+			"INSERT into t values (1, 10, '100');",
+		},
+		LeftSetUpScript: []string{
+			"alter table t add index idx1 (col2);",
+			"INSERT into t values (2, 20, '200');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:       "call dolt_merge('right');",
+				ExpectedErr: merge.ErrSchemaConflict,
```

Comment:
I think this will be a "successful" merge with schema conflicts now

Reply:
Nice! 🙌